jetty-server/src/main/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandler.java

```diff
@@ -0,0 +1,265 @@
+ * </p>
+ * <p>
+ * Note that the decision to buffer is influenced by the headers and status at the
+ * first write, and thus subsequent changes to those headers will not influence the
+ * decision to buffer or not.
+ * </p>
+ * <p>
+ * Note also that there are no memory limits to the size of the buffer, thus
+ * this handler can represent an unbounded memory commitment if the content
+ * generated can also be unbounded.
+ * </p>
+ */
+public class FileBufferedResponseHandler extends BufferedResponseHandler
+{
+    private static final Logger LOG = LoggerFactory.getLogger(FileBufferedResponseHandler.class);
+
+    private Path _tempDir = new File(System.getProperty("java.io.tmpdir")).toPath();
+
+    public Path getTempDir()
+    {
+        return _tempDir;
+    }
+
+    public void setTempDir(Path tempDir)
+    {
+        _tempDir = Objects.requireNonNull(tempDir);
+    }
+
+    @Override
+    protected BufferedInterceptor newBufferedInterceptor(HttpChannel httpChannel, Interceptor interceptor)
+    {
+        return new FileBufferedInterceptor(httpChannel, interceptor);
+    }
+
+    private class FileBufferedInterceptor implements BufferedResponseHandler.BufferedInterceptor
+    {
+        private static final int MAPPED_BUFFER_SIZE = Integer.MAX_VALUE / 2;
+
+        private final Interceptor _next;
+        private final HttpChannel _channel;
+        private Boolean _aggregating;
+        private Path _filePath;
+        private OutputStream _fileOutputStream;
+
+        public FileBufferedInterceptor(HttpChannel httpChannel, Interceptor interceptor)
+        {
+            _next = interceptor;
+            _channel = httpChannel;
+        }
+
+        @Override
+        public void resetBuffer()
+        {
+            if (_filePath != null)
+            {
+                try
+                {
+                    Files.delete(_filePath);
+                }
+                catch (Throwable t)
+                {
+                    LOG.warn("Could Not Delete File {}", _filePath, t);
+                }
+            }
+
+            IO.close(_fileOutputStream);
+            _fileOutputStream = null;
+            _aggregating = null;
+            _filePath = null;
+        }
+
+        @Override
+        public void write(ByteBuffer content, boolean last, Callback callback)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("{} write last={} {}", this, last, BufferUtil.toDetailString(content));
+
+            // If we are not committed, must decide if we should aggregate or not.
+            if (_aggregating == null)
+            {
+                if (last)
+                    _aggregating = Boolean.FALSE;
+                else
+                {
+                    Response response = _channel.getResponse();
+                    int sc = response.getStatus();
+                    if (sc > 0 && (sc < 200 || sc == 204 || sc == 205 || sc >= 300))
```

Comment:
Duplicated logic with `BufferedResponseHandler` is should be factored into a common method.